### ISSUE TITLE
GC error not being reported when using Zinc server

### ISSUE THREAD

**Issue by dragos** [triplequote/scala-maven-plugin#6]
Not sure who is at fault here, maybe Zinc or the JVM. When Hydra discovers that too much GC time is spent while compiling it issues an error (`reporter.error`). This happens inside the JVM notification for GC activity. This error is *never* printed, but the build still fails. Here's what we know:

- this only happens when using Zinc **server**. Everything works outside of Nailgun
- the error is not printed and is not "seen" inside the debugger (breakpoint at Zinc's `LoggingReporter.log`)
- the error count is 1, which means the error has been at least counted
- the error shows up inside `allProblems` at the end of compilation, when hitting a breakpoint inside `LoggingReporter.printSummary`
- breakpoints are not hit on Hydra's side, inside the `GarbageCollection.notification`
- printlns or `System.out/err.println` don't show up anywhere inside the same method, even though the method is called (it's call to `reporter.error` causes the build to fail)

This could be a problem with stdin/stdout redirection inside Nailgun, though that doesn't explain why the breakpoint is not hit (but other breakpoints *are* hit).

**Comment by dragos** [comment]
The mystery is slowly fading away. The output streams are replaced by instances of something that's ThreadLocal, and obviously the thread that runs the GC notification doesn't have a proper thread local for it's output stream.

```
System.out: com.martiansoftware.nailgun.ThreadLocalPrintStream@3e3dcfc5
Console.out: com.martiansoftware.nailgun.ThreadLocalPrintStream@3e3dcfc5
System.err: com.martiansoftware.nailgun.ThreadLocalPrintStream@662bc715
Console.err: com.martiansoftware.nailgun.ThreadLocalPrintStream@662bc715
```

**Comment by dragos** [comment]
Fixed in triplequote/zinc@a79bf80bad6fdd28768ee1ede94bb29ada928a22
